src/main/java/com/mraof/minestuck/entity/item/ConsumableProjectileEntity.java

```diff
@@ -0,0 +1,73 @@
+package com.mraof.minestuck.entity.item;
+
+import com.mraof.minestuck.client.renderer.entity.RendersAsItem;
+import com.mraof.minestuck.item.MSItems;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.item.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.ProjectileItemEntity;
+import net.minecraft.item.Item;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.IPacket;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.EntityRayTraceResult;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.network.NetworkHooks;
+
+public class ConsumableProjectileEntity extends ProjectileItemEntity implements RendersAsItem
+{
+	public ConsumableProjectileEntity(EntityType<? extends ConsumableProjectileEntity> type, World worldIn)
+	{
+		super(type, worldIn);
+	}
+	
+	public ConsumableProjectileEntity(EntityType<? extends ConsumableProjectileEntity> type, double x, double y, double z, World worldIn)
+	{
+		super(type, x, y, z, worldIn);
+	}
+	
+	public ConsumableProjectileEntity(EntityType<? extends ConsumableProjectileEntity> type, LivingEntity livingEntityIn, World worldIn)
+	{
+		super(type, livingEntityIn, worldIn);
+	}
+	
+	@Override
+	protected void onImpact(RayTraceResult result)
+	{
+		if(this.getThrower() instanceof PlayerEntity)
+		{
+			PlayerEntity throwerPlayer = (PlayerEntity) this.getThrower();
+			if(!this.world.isRemote && result.getType() == RayTraceResult.Type.ENTITY)
+			{
+				Entity entity = ((EntityRayTraceResult) result).getEntity();
+				entity.attackEntityFrom(DamageSource.causeThrownDamage(this, this.getThrower()), 2);
+			}
+			if(!throwerPlayer.isCreative() && rand.nextFloat() < 0.99F)
+			{
+				ItemEntity itemEntity = new ItemEntity(this.world, this.getPosX(), this.getPosY(), this.getPosZ(), this.getItem());
+				world.addEntity(itemEntity);
+			}else if(!throwerPlayer.isCreative() && rand.nextFloat() >= 0.99F)
```

Comment:
I don't think it's intentional to generate a new float value here. I'd assume that you want something like
`if(not creative) {if(drop probability) drop item; else play break sound;}`